apiserver/common/secrets/secrets.go

```diff
@@ -152,6 +152,25 @@ func DrainBackendConfigInfo(backendID string, model Model, authTag names.Tag, le
 	return &result, nil
 }
 
+// SecretCleanupBackendConfigInfo returns the config to delete any application owned secrets
+// when the app is removed.
```

Comment:
So this is also used for unit removal?

Reply:
Secrets owned by units are deleted in the unit agent. There's already support for this because the unit agent needs to do it if the charm calls `secret-remove`.